// ngx-wig/esm2022/lib/config.mjs
import { InjectionToken } from '@angular/core';
export const DEFAULT_LIBRARY_BUTTONS = {
    list1: {
        label: 'UL',
        title: 'Unordered List',
        command: 'insertunorderedlist',
        styleClass: 'nw-list-ul',
        icon: 'nwe-icon-list-ul'
    },
    list2: {
        label: 'OL',
        title: 'Ordered List',
        command: 'insertorderedlist',
        styleClass: 'nw-list-ol',
        icon: 'nwe-icon-list-ol'
    },
    bold: {
        label: 'B',
        title: 'Bold',
        command: 'bold',
        styleClass: 'nw-bold',
        icon: 'nwe-icon-bold'
    },
    italic: {
        label: 'I',
        title: 'Italic',
        command: 'italic',
        styleClass: 'nw-italic',
        icon: 'nwe-icon-italic'
    },
    link: {
        label: 'Link',
        title: 'Link',
        command: 'createlink',
        styleClass: 'nw-link',
        icon: 'nwe-icon-link'
    },
    underline: {
        label: 'U',
        title: 'Underline',
        command: 'underline',
        styleClass: 'nw-format-underlined',
        icon: 'nwe-icon-underline'
    }
};
export const CUSTOM_LIBRARY_BUTTONS = {
    edithtml: {
        label: 'Edit HTML',
        title: 'Edit HTML',
        command: (ctx) => {
            ctx.editMode = !ctx.editMode;
        },
        styleClass: 'nw-button--source',
        icon: '',
    }
};
export const BUTTONS = new InjectionToken('BUTTONS');
